### DISCUSSION TITLE
netCDF file Unrecognized

### DISCUSSION THREAD

**Discussion by drcjmelick** [dtcenter/METplus#1137] [File I/O] [answered]
Hi all,

I'm having problems with a netCDF file I'm trying to use in regrid_data_plane not being recognized.   What is the excepted format for the header and data contents in the netCDF usage with MET?   Thanks!

Chris

**Comment by rvalenzuelar** [comment]
I have an issue with a custom-made CF-compliance NetCDF file as well. I wonder if anyone here can share functional examples of NetCDF files to copy the structure. In my case is a GFS forecast file.

**Comment by j-opatz** [comment, marked as answer]
Hi Chris, and thank you for your question. 

As you probably already read in the MET User's [Guide](https://met.readthedocs.io/en/latest/Users_Guide/data_io.html#input-data-formats), MET can handle "NetCDF files following the Climate and Forecast (CF) conventions, containing WRF output post-processed using wrf_interp, or produced by the MET tools themselves". 
While I'm not sure that MET has example netCDF files, I do know that METplus has [tarballs](https://dtcenter.ucar.edu/dfiles/code/METplus/METplus_Data/v4.0/) of data for each of its use cases by category, which includes netCDFs. Downloading one or more of those tarballs and reviewing the netCDFs would be a good start.
Additionally, MET is capable of reading in netCDFs via a Python script: take a look at [this](https://metplus.readthedocs.io/en/latest/generated/model_applications/air_quality_and_comp/EnsembleStat_fcstICAP_obsMODIS_aod.html#python-embedding) Python embedded script as an example. As you can see in the `attrs` dictionary, a certain amount of information needs to be available to MET for proper interpretation.
If after reviewing this material you're still having trouble with your netCDF file, you're welcome to perform a header dump of it here and we can take a look.

**Comment by drcjmelick** [comment]
I appreciate the detailed response and the options you gave me to determine what I need to add to the netCDF file to make it compliant with what MET expects.   I will give it a try and respond if I still am having issues!

Thanks,
Chris

From: j-opatz ***@***.***>
Sent: Friday, September 3, 2021 8:55 AM
To: dtcenter/METplus ***@***.***>
Cc: MELICK, CHRISTOPHER J GS-12 USAF ACC 16 WS/WXD ***@***.***>; Author ***@***.***>
Subject: [Non-DoD Source] Re: [dtcenter/METplus] netCDF file Unrecognized (#1137)


Hi Chris, and thank you for your question.

As you probably already read in the MET User's Guide<https://met.readthedocs.io/en/latest/Users_Guide/data_io.html#input-data-formats>, MET can handle "NetCDF files following the Climate and Forecast (CF) conventions, containing WRF output post-processed using wrf_interp, or produced by the MET tools themselves".
While I'm not sure that MET has example netCDF files, I do know that METplus has tarballs<https://dtcenter.ucar.edu/dfiles/code/METplus/METplus_Data/v4.0/> of data for each of its use cases by category, which includes netCDFs. Downloading one or more of those tarballs and reviewing the netCDFs would be a good start.
Additionally, MET is capable of reading in netCDFs via a Python script: take a look at this<https://metplus.readthedocs.io/en/latest/generated/model_applications/air_quality_and_comp/EnsembleStat_fcstICAP_obsMODIS_aod.html#python-embedding> Python embedded script as an example. As you can see in the attrs dictionary, a certain amount of information needs to be available to MET for proper interpretation.
If after reviewing this material you're still having trouble with your netCDF file, you're welcome to perform a header dump of it here and we can take a look.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/dtcenter/METplus/discussions/1137#discussioncomment-1277272>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AVOWTOZ7YXHEJGUOQAVIUM3UADHTRANCNFSM5DKAXPGQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
